Achievements/SelfMade.lua

```diff
@@ -1,53 +1,92 @@
 local _G = _G
 local self_made_achievement = CreateFrame("Frame")
 _G.achievements.SelfMade = self_made_achievement
+local player = UnitName("Player")
 
 -- General info
 self_made_achievement.name = "SelfMade"
 self_made_achievement.title = "Self-Made"
 self_made_achievement.class = "All"
 self_made_achievement.icon_path = "Interface\\Addons\\Hardcore\\Media\\icon_self_made.blp"
 self_made_achievement.description =
-	"Complete the Hardcore challenge without at any point equipping an item that you have not crafted yourself. Items your character has conjured (e.g. Firestones) are considered crafted. No items bought, dropped, or rewarded by quests are allowed to be equipped (items provided for a quest can be equipped). The items your character starts with are allowed to be equipped. Bags are equipped items."
+"Complete the Hardcore challenge without at any point equipping an item that you have not crafted yourself. Items your character has conjured (e.g. Firestones) are considered crafted. No items bought, dropped, or rewarded by quests are allowed to be equipped, fishing poles MAY be equipped, starting items MAY be re-equipped if taken off (items provided for a quest can be equipped). The items your character starts with are allowed to be equipped. Bags are equipped items."
+local Combine = {
+    35, 36, 38, 39, 40, 43, 44, 45, 47, 48, 49, 51, 52, 53, 55, 56, 57, 59, 120, 121, 127, 129, 139, 140, 147, 148, 153,
+    154, 1395, 1396, 2092, 2105, 2361, 2362, 2504, 2508, 2512, 2516, 2947, 3111, 3661, 6096, 6097, 6098, 6117, 6118, 6119,
+    6120, 6121, 6122, 6123, 6124, 6125, 6126, 6127, 6129, 6134, 6135, 6136, 6137, 6138, 6139, 6140, 6144, 12282, 20891,
+    20892, 20893, 20894, 20895, 20896, 20897, 20898, 20899, 20900, 20901, 20978, 20982, 23322, 23344, 23345, 23346, 23347,
+    23348, 23473, 23474, 23475, 23476, 23477, 23478, 23479, 24143, 24145, 24146, 25861, 28979, 49778, 50055, 50057
+}
+
 
 -- Registers
 function self_made_achievement:Register(fail_function_executor)
-	self_made_achievement:RegisterEvent("PLAYER_EQUIPMENT_CHANGED")
-	self_made_achievement.fail_function_executor = fail_function_executor
+    self:RegisterEvent("PLAYER_EQUIPMENT_CHANGED")
+    self:RegisterEvent("ITEM_UNLOCKED")
+    self.fail_function_executor = fail_function_executor
 end
 
 function self_made_achievement:Unregister()
-	self_made_achievement:UnregisterEvent("PLAYER_EQUIPMENT_CHANGED")
+    self:UnregisterEvent("PLAYER_EQUIPMENT_CHANGED")
+    self:RegisterEvent("ITEM_UNLOCKED")
+end
+
+local function Start(a)
+    local item_id = GetInventoryItemID("player", a)
+    for index, value in ipairs(Combine) do
+        if string.match(item_id, value) then
+            GameTooltip:Hide()--prevents a hung empty tooltip window
```

Comment:
suprised this is needed here (and not just in isSelfCreated) but it doesn't hurt

Reply:
I generally break out my functions to prevent the code from getting murky with too much logic in 1 function.